### DISCUSSION TITLE
Procedure to upgrade from 3.1.2 to 3.2.3

### DISCUSSION THREAD

**Discussion by drbeco** [PodcastGenerator/PodcastGenerator#672] [Q&A] [answered]
Please, what is the procedure to upgrade the podcast?

I have it under `/www/public_html`. I tried:

```
/www $ git clone git@github.com:PodcastGenerator/PodcastGenerator.git
/www/PodcastGenerator $ git checkout v3.2.3
```

And then i tried two ways:

```
/www $ rsync -ahu -vxHAX -i --stats --update PodcastGenerator/PodcastGenerator/ public_html/
```

Didn't work. I tried a simple:

```
/www $ mv public_html public_html-bkp
/www $ cp -a PodcastGenerator/PodcastGenerator public_html
```

Also to no avail. Any help appreciated.

**Comment by charabaruk** [comment, marked as answer]
If you are cloning from Git, then you need to make sure that you have Composer installed, and then do `composer install` inside the PodcastGenerator/ directory in your working copy. If you are using the pre-built zips (best way to use PG if you are not developing on it) then you should be able to extract the PodcastGenerator/ directory over top of your existing PG instance as per the [upgrade instructions](https://github.com/PodcastGenerator/PodcastGenerator/blob/master/doc/release-notes/3.2.3.md#upgrading-from-previous-versions) in the release notes.

**Comment by charabaruk** [comment]
I probably should provide a link to Composer as well: https://getcomposer.org/

**Comment by charabaruk** [comment]
Finally, you need to be using PHP 7.0 or newer (8.0 or newer preferred, especially as all versions older than 7.4 are completely unsupported by the PHP devs).

**Comment by drbeco** [comment]
Not that easy with Slackware Linux. Is there a simple way to just update, without Composer?

**Reply by charabaruk** [reply]
Not if you're using Git instead of downloading the official release zips. The release zips come with all Composer dependencies already included.

**Reply by drbeco** [reply]
How do I upgrade from the release zips ? I think this is a better approach for me.

Can I unzip it over the current folder?
